src/Nethermind/Nethermind.TxPool/TxPool.NonceCompositeComparer.cs

```diff
@@ -0,0 +1,52 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core;
+
+namespace Nethermind.TxPool
+{
+    public partial class TxPool
+    {
+        /// <summary>
+        /// Orders first by <see cref="Transaction.Nonce"/> asc and then by inner comparer
+        /// </summary>
+        public class NonceCompositeComparer : IComparer<Transaction>
```

Comment:
This is a Decorator and not a Composite
hmmm why DefaultComparer is not by nonce?

Reply:
Because I want to use them differently in some cases. Also I allow to provide comparer to some components like TxPool and in TxPool I decorate it because TxPool needs Nonce comparision first and then other Tx comparision. We cannot rely that nonce will be highest order in provided comparator.

Added additional comments in every place decorators are used.